tools/cldr-code/src/main/java/org/unicode/cldr/util/MissingXmlGetter.java

```diff
@@ -0,0 +1,35 @@
+package org.unicode.cldr.util;
+
+import java.io.*;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class MissingXmlGetter {
+
+    private final CLDRFile cldrFile;
+
+    public MissingXmlGetter(CLDRLocale loc, Factory factory) {
+        this.cldrFile = factory.make(loc.getBaseName(), true/* resolved */);
+    }
+
+    /**
+     * Construct XML for the paths that are error/missing/provisional in a locale, with value placeholders,
+     * so that somebody can download and edit the .xml file to fill in the values and then do a bulk
+     * submission of that file
+     *
+     * @param coverageLevel the coverage level for paths to be included
+     * @return the XML as a string
+     * @throws IOException for StringWriter
+     */
+    public String getXml(Level coverageLevel) throws IOException {
+        try (StringWriter sw = new StringWriter(); PrintWriter pw = new PrintWriter(sw)) {
+            // TODO: add a new option telling cldrFile.write to write only the error/missing/provisional paths
+            // (at the given coverageLevel, adding a placeholder for the values) instead of
+            // the present paths
```

Comment:
the new option should be added in the next PR

this PR refactors CLDRFile.write by moving a lot of code; the change in functionality will be separated

Reply:
CLDRFile won't know which paths are errors.  Would it be cleaner to start with an empty CLDRFile (XMLSource) and only add the paths which are M/E/P, and explicitly set any comments?